pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -160,23 +160,26 @@ func isExpectedGRPCError(err error) bool {
 func receiveThread(con *XdsConnection, reqChannel chan *discovery.DiscoveryRequest, errP *error) {
 	defer close(reqChannel) // indicates close of the remote side.
 	for {
+		con.mu.RLock()
```

Comment:
Is this needed? I don't see where we mutate ConID so I am a bit surprised?

Reply:
Apparently we do - according the the race test detector. There is another thread that sets it on connect, and it seems a timing issue

Reply:
Ah got it, I was thinking you added a new place that mutates it but instead probably triggered the condition